OrbitQt/orbitsamplingreport.cpp

```diff
@@ -86,6 +89,21 @@ void OrbitSamplingReport::on_PreviousCallstackButton_clicked() {
   RefreshCallstackView();
 }
 
+void OrbitSamplingReport::OnCurrentThreadTabChanged(int current_tab_index) {
+  if (current_tab_index == -1) {
```

Comment:
When can this case happen? Looks like this should also clear the selection?

Reply:
This happens when the user click on the same thread tab. 

To be more specific, we receive the current_tab_index from the signal currentChanged every time when we select a thread tab. If the selected tab is the same with previous one, current_tab_index is -1; otherwise, current_tab_index illustrates the page index of selected new thread tab.

Reply:
But it might be a good idea that we consider the user can click on the same thread tab to clear the previous selection. This might be helpful when for the multiple selection case.

Reply:
Are you sure about the -1 being sent when the same tab is clicked? That would be super ugly. The way I read the documentation, -1 is passed in exceptional cases, e.g. when the TabWidget does not have any tabs that could be selected. If the same tab is clicked, this signal shouldn't be emitted at all.

Which means clearing the selection would be the right thing to do for -1, even though we do not expect this to happen.

Reply:
I checked the qt5 document for [QTabWidget::currentChanged](https://doc.qt.io/qt-5/qtabwidget.html#currentChanged). And I tested on the ui and found that if I click on the same thread tab, current_tab_index is -1.